File: internal/content/testdata/bench/csharp/product-of-letter-counts.cs
using System.Collections.Generic;

public class Solution {
    public long product_of_counts(string s) {
        const long MOD = 1_000_000_007;
        var freq = new Dictionary<char, long>();
        foreach (var ch in s) {
            if (!freq.ContainsKey(ch)) freq[ch] = 0;
            freq[ch]++;
        }
        long result = 1;
        foreach (var cnt in freq.Values) {
            result = (result * cnt) % MOD;
        }
        return result;
    }
}

File: internal/content/testdata/bench/go/product-of-letter-counts.go
func product_of_counts(s string) int {
	const MOD = 1_000_000_007
	freq := make(map[rune]int)
	for _, ch := range s {
		freq[ch]++
	}
	result := 1
	for _, cnt := range freq {
		result = (result * cnt) % MOD
	}
	return result
}
